### PR TITLE
Guard version.h macros from SWIG parser

### LINKED ISSUE #4355: swig cannot parse CGAL/version.h
swig fails to parse version.h:
```
#define CGAL_VERSION 4.14.2
#define CGAL_VERSION_NR 1041421000
#define CGAL_SVN_REVISION 99999
#define CGAL_GIT_HASH 19efe6e568f8d9cf090193b029b143e769b3ea18
#define CGAL_RELEASE_DATE 20190329
```
it chokes on the CGAL_GIT_HASH define with ":EOF: Error: Exponent does not have any digits"
I'll ask swig (https://github.com/swig/swig/issues/1666), but you may want to add quotes here

The version number may be quoted too to avoid the "Bad constant value" warning

### PR DESCRIPTION
**Pull request by xantares**
Closes #4355

### CODE REVIEW

**Comment by lrineau** [comment]
Please @sloriot do not merge.

**Comment by lrineau** [comment]
I am against that change because it might break actual usage of `CGAL_VERSION` and `CGAL_GIT_HASH`. Is not there another way to please the SWIG parser?

**Comment by sloriot** [comment]
I looked a bit and could not find a workaround. Easiest solution: I could stop using this file but it means dropping the support of older versions of CGAL.

**Comment by xantares** [comment]
the unquoted CGAL_VERSION and CGAL_GIT_HASH macros in version.h are not even valid c++
if you put them in a minimal main.cxx you will see that g++ will return these on of these two errors:
```
main.cxx:3:22: error: too many decimal points in number
 #define CGAL_VERSION 4.14.2
                                          ^~~~~~~
main.cxx:4:23: error: exponent has no digits
 #define CGAL_GIT_HASH 19efe6e568f8d9cf090193b029b143e769b3ea18
                                            ^
```

**Comment by lrineau** [comment]
[**@xantares**](https://github.com/xantares) commented on [Nov 26, 2019, 8:49 AM GMT+1](https://github.com/CGAL/cgal/pull/4358#issuecomment-558505256 "2019-11-26T07:49:30Z - Replied by Github Reply Comments"):
> the unquoted CGAL_VERSION and CGAL_GIT_HASH macros in version.h are not even valid c++
> if you put them in a minimal main.cxx you will see that g++ will return one of these errors:
> ```
> main.cxx:3:22: error: too many decimal points in number
>  #define CGAL_VERSION 4.14.2
>                       ^~~~~~~
> main.cxx:4:23: error: exponent has no digits
>  #define CGAL_GIT_HASH 19efe6e568f8d9cf090193b029b143e769b3ea18
>                        ^
> ```
> the other macros are fine because they are just integers

That is why we have the macro `CGAL_STR` in `<CGAL/version_macros.h>`, to stringify the token. There is even `CGAL_VERSION_STR`, that is defined as `CGAL_STR(CGAL_VERSION)`.

**Comment by xantares** [comment]
So CGAL_VERSION must be quoted and deprecated for CGAL_VERSION_STR

**Comment by lrineau** [comment]
We cannot do so without breaking the compatibility with existing code.

If the definition of `CGAL_VERSION` and `CGAL_GIT_HASH` are problematic for the Swig parser, then I suggest to surround them with:
```c++
#ifndef SWIG
// ...
#endif // SWIG
```

**Comment by xantares** [comment]
@lrineau @sloriot see the new patch here

**Comment by sloriot** [comment]
@xantares It is not going to work, the current version of the bindings needs to parse the file in order to determine if some functionalities are available.

**Comment by lrineau** [comment]
What if only `CGAL_VERSION` and `CGAL_GIT_HASH` where inside `#ifndef SWIG`, with `CGAL_VERSION_NR` and others outside?

**Comment by xantares** [comment]
@sloriot I dont see where the bindings need the version macro from the header

**Comment by sloriot** [comment]
```
ack -l  CGAL_VERSION ~/projects/cgal-swig-bindings.git/ 
/home/sloriot/projects/cgal-swig-bindings.git/SWIG_CGAL/Point_set_processing_3/functions.h
/home/sloriot/projects/cgal-swig-bindings.git/SWIG_CGAL/Mesh_3/CGAL_Mesh_3.i
/home/sloriot/projects/cgal-swig-bindings.git/SWIG_CGAL/Triangulation_3/Regular_triangulation_3.h
/home/sloriot/projects/cgal-swig-bindings.git/SWIG_CGAL/Triangulation_3/CGAL_Triangulation_3.i
/home/sloriot/projects/cgal-swig-bindings.git/SWIG_CGAL/Triangulation_3/typedefs.h
/home/sloriot/projects/cgal-swig-bindings.git/SWIG_CGAL/Kernel/Weighted_point_3.h
/home/sloriot/projects/cgal-swig-bindings.git/SWIG_CGAL/Kernel/Weighted_point_2.h
/home/sloriot/projects/cgal-swig-bindings.git/SWIG_CGAL/Mesh_2/CGAL_Mesh_2.i
/home/sloriot/projects/cgal-swig-bindings.git/SWIG_CGAL/Triangulation_2/Constrained_triangulation_plus_2.h
/home/sloriot/projects/cgal-swig-bindings.git/SWIG_CGAL/Triangulation_2/CGAL_Triangulation_2.i
/home/sloriot/projects/cgal-swig-bindings.git/SWIG_CGAL/Triangulation_2/Triangulation_2.h
/home/sloriot/projects/cgal-swig-bindings.git/SWIG_CGAL/Triangulation_2/Regular_triangulation_2.h
/home/sloriot/projects/cgal-swig-bindings.git/SWIG_CGAL/Triangulation_2/typedefs.h
/home/sloriot/projects/cgal-swig-bindings.git/SWIG_CGAL/Advancing_front_surface_reconstruction/impl.h
/home/sloriot/projects/cgal-swig-bindings.git/SWIG_CGAL/Alpha_shape_2/CGAL_Alpha_shape_2.i
/home/sloriot/projects/cgal-swig-bindings.git/SWIG_CGAL/Alpha_shape_2/typedefs.h
/home/sloriot/projects/cgal-swig-bindings.git/SWIG_CGAL/Polygon_mesh_processing/CGAL_Polygon_mesh_processing.i
```

**Comment by xantares** [comment]
@sloriot  we only use CGAL_VERSION_NR, so @lrineau 's proposition would work

File: Installation/include/CGAL/version.h [hunk 1]
```diff
@@ -16,7 +16,10 @@
 #ifndef CGAL_VERSION_H
 #define CGAL_VERSION_H
 
+#ifndef SWIG
 #define CGAL_VERSION 5.1
+#define CGAL_GIT_HASH abcdef
+#endif
 #define CGAL_VERSION_NR 1050100000
 #define CGAL_SVN_REVISION 99999
 #define CGAL_GIT_HASH abcdef
```
Comment (sloriot) [inline Installation/include/CGAL/version.h]:
```suggestion
```

Comment (xantares) [reply Installation/include/CGAL/version.h]: done

**Comment by lrineau** [comment]
OK, let's try to get this merged into CGAL for CGAL-5.1.

**Comment by maxGimeno** [comment]
We'll try after the releases bug fixes

**Comment by sloriot** [comment]
It should go into both release branches otherwise it will be broken for a while in conda AFAIU

**Comment by lrineau** [comment]
@xantares Can you please give us (@lrineau, @sloriot, @maxGimeno) the rights to edit your pull-request? We would like to rebase it to the branch for 4.14.x.

**Comment by maxGimeno** [comment]
This PR breaks the testsuites windows that are not header only. The reason is that [thos file](https://github.com/CGAL/cgal/blob/3b8d77aa8a2570cf599a9d41918809947632842b/Scripts/developer_scripts/cgal_create_release_with_cmake.cmake) edits version.h using a regex that won't work with that change. You need to update those REGEX too.

**Comment by lrineau** [comment]
I have replaced this PR with https://github.com/CGAL/cgal/pull/4475, to be able to add a commit to it.

**Pull request closed without merging**